Refactor isAuthorized utility function

geichelberger opened the pull request:
It's just a minor refactoring, which brings back the old function signature and rearranges the code in the for loop.
The refactoring of the for loop was negotiating the if statement and moving the "else" part into the body, significantly simplifying the code.

Improve readability by:

- Simplifying nested if conditions
- Removing unnecessary continue statements
- Adding overloaded function to support optional parameters

### To be discussed

Do we need the `episodeRoleId` argument? Wouldn't the `mediapackageId` be sufficient?
If it's null, we skip the check.

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)
* [ ] explain why it needs to be merged into the legacy branch, if it is targeting the legacy branch

geichelberger committed:
Refactor isAuthorized function

Improve readability by:

- Simplifying nested if conditions
- Removing unnecessary continue statements
- Adding overloaded function to support optional parameters

gregorydlogan reviewed (approved):
You're correct that `episodeRoleId` can go.  Ping me when you're pulled it out and we can get this merged.

geichelberger committed:
Remove episodeRoleId argument

Instead of having an argument which indicated that we want to check the
ACLs against mediaPakcageId role. We check if it's not null and then do
the check.

geichelberger commented:
@gregorydlogan, I pulled out the argument, and it's ready for review.
